src/components/panel/panel.js

```diff
@@ -1128,46 +1128,35 @@ MdPanelRef.prototype._createPanel = function() {
 
 /**
  * Adds the styles for the panel, such as positioning and z-index.
- * @return {!angular.$q.Promise}
+ * @return {!angular.$q.Promise<MdPanelRef>}
  * @private
  */
 MdPanelRef.prototype._addStyles = function() {
-  this._panelContainer.css('z-index', this._config['zIndex']);
-  this._panelEl.css('z-index', this._config['zIndex'] + 1);
-
-
-  if (this._config['fullscreen']) {
-    this._panelEl.addClass('_md-panel-fullscreen');
-    return this._$q.resolve(this); // Don't setup positioning.
-  }
-
-  return this._configurePosition();
-};
+  var self = this;
+  return this._$q(function(resolve) {
+    self._panelContainer.css('z-index', self._config['zIndex']);
+    self._panelEl.css('z-index', self._config['zIndex'] + 1);
 
+    if (self._config['fullscreen']) {
+      self._panelEl.addClass('_md-panel-fullscreen');
+      return resolve(self); // Don't setup positioning.
+    }
 
-/**
- * Configure the position of the panel.
- * @return {!angular.$q.Promise}
- * @private
- */
-MdPanelRef.prototype._configurePosition = function() {
-  var positionConfig = this._config['position'];
-  if (!positionConfig) {
-    return this._$q.resolve(this);
-  }
+    var positionConfig = self._config['position'];
+    if (!positionConfig) {
+      return resolve(self); // Don't setup positioning.
+    }
 
-  var self = this;
-  return this._$q(function(resolve) {
     // Wait for angular to finish processing the template, then position it
-    // correctly. This is necessary so that the panel will have a defined
-    // height and width.
-    return self._$rootScope['$$postDigest'](function () {
+    // correctly. This is necessary so that the panel will have a defined height
+    // and width.
+    self._$rootScope['$$postDigest'](function() {
       self._panelEl.css('top', positionConfig.getTop(self._panelEl));
       self._panelEl.css('bottom', positionConfig.getBottom(self._panelEl));
       self._panelEl.css('left', positionConfig.getLeft(self._panelEl));
       self._panelEl.css('right', positionConfig.getRight(self._panelEl));
       self._panelEl.css('transform', positionConfig.getTransform());
```

Comment:
I think, using the `$mdConstant` is doing the same.

- https://github.com/angular/material/blob/master/src/core/util/constant.js#L39